tests/integration/test_base_service.py

```diff
@@ -116,6 +116,36 @@ def test_base_service_process_request_test_scenario_replays_ok(dispatcher):
     assert response_service_2.body == b'test_scenario_replays_ok/TestServiceB/0'
 
 
+def test_base_service_process_request_test_scenario_picks_next_after_reaches_max_replay(dispatcher):
```

Comment:
I would also add a test to check other numbers than `max_replay: 1` for good measure :slightly_smiling_face: